src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8.cs

```diff
@@ -815,5 +824,192 @@ private bool Fail()
         /// <returns><c>true</c> if value is well-formed UTF-8, <c>false</c> otherwise.</returns>
         public static bool IsValid(ReadOnlySpan<byte> value) =>
             Utf8Utility.GetIndexOfFirstInvalidUtf8Sequence(value, out _) < 0;
+
+#if !NET
+        /// <summary>
+        /// Decodes the Rune at the beginning of the provided UTF-8 source buffer.
+        /// </summary>
+        /// <returns>
+        /// <para>
+        /// If the source buffer begins with a valid UTF-8 encoded scalar value, returns <see cref="OperationStatus.Done"/>,
+        /// and outs via <paramref name="result"/> the decoded Runeand via <paramref name="bytesConsumed"/> the
+        /// number of <see langword="byte"/>s used in the input buffer to encode the Rune.
+        /// </para>
+        /// <para>
+        /// If the source buffer is empty or contains only a partial UTF-8 subsequence, returns <see cref="OperationStatus.NeedMoreData"/>,
+        /// and outs via <paramref name="result"/> ReplacementChar and via <paramref name="bytesConsumed"/> the length of the input buffer.
+        /// </para>
+        /// <para>
+        /// If the source buffer begins with an ill-formed UTF-8 encoded scalar value, returns <see cref="OperationStatus.InvalidData"/>,
+        /// and outs via <paramref name="result"/> ReplacementChar and via <paramref name="bytesConsumed"/> the number of
+        /// <see langword="char"/>s used in the input buffer to encode the ill-formed sequence.
+        /// </para>
+        /// </returns>
+        /// <remarks>
+        /// The general calling convention is to call this method in a loop, slicing the <paramref name="source"/> buffer by
+        /// <paramref name="bytesConsumed"/> elements on each iteration of the loop. On each iteration of the loop <paramref name="result"/>
+        /// will contain the real scalar value if successfully decoded, or it will contain ReplacementChar if
+        /// the data could not be successfully decoded. This pattern provides convenient automatic U+FFFD substitution of
+        /// invalid sequences while iterating through the loop.
+        /// </remarks>
+        private static OperationStatus DecodeFromUtf8(ReadOnlySpan<byte> source, out uint result, out int bytesConsumed)
```

Comment:
Can the project include Rune.cs instead of including a copy of the implementation here?

Reply:
Done in https://github.com/teo-tsirpanis/dotnet-runtime/commit/79ee05d67f31c0dd7b218c2cf6b2c864b96ce8db in order to unblock work that depends on this, after waiting for one week. Anyone with permissions can push my commit on the PR's branch.

Reply:
pushed

Reply:
Compile errors fixed in https://github.com/dotnet/runtime/commit/bf6f989f91e88ea75a07e4870faf98a8fb711b88.

Reply:
pushed